src/NLog/Internal/TargetWithFilterChain.cs

```diff
@@ -92,17 +94,21 @@ internal void PrecalculateStackTraceUsage()
 
             // find all objects which may need stack trace
             // and determine maximum
-            foreach (var item in ObjectGraphScanner.FindReachableObjects<IUsesStackTrace>(this))
+            // only the target can have IUsesStackTrace
+            if (Target != null)
```

Comment:
Don't you need to recurse NextInChain as well? In case those LayoutRenderers require stacktraces?